lib/linked_list.rb

```diff
@@ -89,34 +233,55 @@ def has_cycle
       raise NotImplementedError
     end
 
-
-    # Additional Exercises 
-    # returns the value in the first node
-    # returns nil if the list is empty
+    # # Additional Exercises 
+    # # returns the value in the first node
+    # # returns nil if the list is empty
     def get_first
-      raise NotImplementedError
+      if @head == nil
+        return nil
+      else
+        return @head.value
+      end
     end
 
-    # method that inserts a given value as a new last node in the linked list
+    # # method that inserts a given value as a new last node in the linked list
     def add_last(value)

```

Comment:
I realize it's not fully implemented, yet, but once `@tail` is working in all the methods, you could just do this:

```ruby
if @head.nil?
  @head = @tail = Node.new(value)
else
  @tail.next = Node.new(value)
  @tail = @tail.next
end
```